src/soep_preparation/utilities/error_handling.py

```diff
@@ -0,0 +1,93 @@
+"""Error handling utilities for data validation."""
+
+from typing import Any
+
+import pandas as pd
+
+
+def _fail_if_series_wrong_dtype(series: pd.Series, expected_dtype: str) -> None:
+    if expected_dtype not in series.dtype.name and expected_dtype != "Any":
+        msg = f"Expected dtype {expected_dtype}, got {series.dtype.name}"
+        raise TypeError(msg)
+
+
+def fail_if_invalid_input(input_: Any, expected_dtype: str) -> None:  # noqa: ANN401
+    """Fail if input is not of expected type.
+
+    Args:
+        input_: The input to check.
+        expected_dtype: The expected type of the input.
+
+    Raises:
+        TypeError: If input is not of expected type.
+    """
+    if expected_dtype not in str(type(input_)) and expected_dtype != "Any":
+        msg = f"Expected {input_} to be of type {expected_dtype}, got {type(input_)}"
+        raise TypeError(
+            msg,
+        )
+
+
+def fail_if_invalid_inputs(input_: Any, expected_dtypes: str) -> None:  # noqa: ANN401
+    """Fail if input is not of any of expected types.
+
+    Args:
+        input_: The input to check.
+        expected_dtypes: The expected types of the input.
+
+    Raises:
+        TypeError: If input is not of any of expected types.
+    """
+    if " | " in expected_dtypes:
+        if not any(
+            expected_dtype in str(type(input_))
+            for expected_dtype in expected_dtypes.split(" | ")
+        ):
+            msg = (
+                f"Expected {input_} to be of type {expected_dtypes}, got {type(input_)}"
+            )
+            raise TypeError(
+                msg,
+            )
+
+    else:
+        fail_if_invalid_input(input_, expected_dtypes)
+
+
+def error_handling_sr_transformation(
```

Comment:
```suggestion
def fail_if_sr_cannot_be_transformed(
```
?

My guess might be wrong, but you'll get the idea.